runtime/src/main/java/org/corfudb/runtime/view/TableRegistry.java

```diff
@@ -256,34 +251,10 @@ void registerTable(@Nonnull String namespace,
                 if (oldRecord == null) { // Case 1 above, new unseen table
                     this.registryTable.put(tableNameKey,
                         new CorfuRecord<>(tableDescriptors, metadataBuilder.build()));
-                } else {
```

Comment:
how do we plan to handle the case of schema change on upgrade?

Reply:
Upgrade needs explicit APIs, we should try to handle it transparently. There could be correctness issues if we do. For example, when two JVMs using two different versions of the same table try to write a key transactionally, the sequencer wont detect conflicts because the serialization is different across versions.